opencti-platform/opencti-front/src/private/components/settings/users/Root.jsx

```diff
@@ -46,72 +42,73 @@ const subscription = graphql`
   }
 `;
 
-class RootUser extends Component {
-  constructor(props) {
-    super(props);
-    const {
-      match: {
-        params: { userId },
-      },
-    } = props;
-    this.sub = requestSubscription({
+const RootUserComponent = ({ queryRef, userId, refetch }) => {
+  const subConfig = useMemo(
+    () => ({
       subscription,
       variables: { id: userId },
-    });
-  }
+    }),
+    [userId],
+  );
+  useSubscription(subConfig);
+  const data = usePreloadedQuery(userQuery, queryRef);
+  const { user } = data;
 
-  componentWillUnmount() {
-    this.sub.dispose();
-  }
+  return (
+    <Security needs={[SETTINGS_SETACCESSES]}>
+      {user ? (
+        <Switch>
+          <Route
+            exact
+            path="/dashboard/settings/accesses/users/:userId"
+            render={(routeProps) => (
+              <User {...routeProps} userData={user} refetch={refetch}/>
+            )}
+          />
+        </Switch>
+      ) : (
+        <ErrorNotFound />
+      )}
+    </Security>
+  );
+};
 
-  render() {
-    const {
-      match: {
-        params: { userId },
-      },
-    } = this.props;
-    return (
-      <Security needs={[SETTINGS_SETACCESSES]}>
-        <TopBar />
-        <QueryRenderer
-          query={userQuery}
-          variables={{
-            id: userId,
-            rolesOrderBy: 'name',
-            rolesOrderMode: 'asc',
-            groupsOrderBy: 'name',
-            groupsOrderMode: 'asc',
-            organizationsOrderBy: 'name',
-            organizationsOrderMode: 'asc',
-          }}
-          render={({ props }) => {
-            if (props) {
-              if (props.user) {
-                return (
-                  <div>
-                    <Route
-                      exact
-                      path="/dashboard/settings/accesses/users/:userId"
-                      render={(routeProps) => (
-                        <User {...routeProps} user={props.user} />
-                      )}
-                    />
-                  </div>
-                );
-              }
-              return <ErrorNotFound />;
-            }
-            return <Loader />;
-          }}
-        />
-      </Security>
-    );
-  }
-}
+const RootUser = () => {
+  const { userId } = useParams();
+  const queryParams = {
+    id: userId,
+    rolesOrderBy: 'name',
+    rolesOrderMode: 'asc',
+    groupsOrderBy: 'name',
+    groupsOrderMode: 'asc',
+    organizationsOrderBy: 'name',
+    organizationsOrderMode: 'asc' };
+  const [queryRef, loadQuery] = useQueryLoader(userQuery);
+  useEffect(() => {
+    loadQuery(queryParams, { fetchPolicy: 'store-and-network' });
+  }, []);
+
+  const refetch = () => {
+    loadQuery(queryParams, { fetchPolicy: 'store-and-network' });
+  };
+
+  return (
+    <div>
+      <TopBar />
+      {queryRef ? (
```

Comment:
You can simplify with 

      {queryRef && (
        <React.Suspense fallback={<Loader variant={LoaderVariant.inElement} />}>
          <RootUserComponent queryRef={queryRef} userId={userId} refetch={refetch}/>
        </React.Suspense>
      )}

Reply:
why ? it's not exactly the same thing because if queryRef is undefined, there will be no loader. And I can see that it's been done like this almost everywhere, even in recent files. Ex: `/cases/case_rfi/Root.tsx`